zproject/prod_settings_template.py

```diff
@@ -211,6 +212,139 @@
+#     nq+PHtGX/hepIM8GBIwsr3oPFiwJBuIX3oduM92mobbTU4GAGCrMLafSArk2ZcQw
+#     y0xP9VuHxvrOVKO+bILZGcslZ74IvvFBderW3hyjAoGBAMEkcZLfe2+FmAe2fju0
+#     F1qCAWqnqpVAgWsuDycS6k5TYHj/6vm8XVx+Xa8r/IfTuyXlAq/CPb4KPPY4ZAUy
+#     iAHVqzslbBuBOHaukySepy2Z4GBaMoD742H0QBdaNQE8pBjZhyir+paTbkehd5kT
+#     8LJ2ctg7ipCEcGe5b+gEllTe
+#     """.replace(' ', '').replace('\n', '').strip()
+
+# SOCIAL_AUTH_SAML_ORG_INFO =  {
+#     "en-US": {
+#         "name": "example",
+#         "displayname": "Example Inc.",
+#         "url": "%s%s" % ('https://', EXTERNAL_HOST),
+#     }
+# }
+
+# SOCIAL_AUTH_SAML_TECHNICAL_CONTACT = {
+#     "givenName": "Tech Gal",
+#     "emailAddress": "technical@example.com"
+# }
+
+# SOCIAL_AUTH_SAML_SUPPORT_CONTACT = {
+#     "givenName": "Support Guy",
+#     "emailAddress": "support@example.com",
+# }
+
+# Important note: Only one IdP is currently supported, don't add
+# any extra IdPs in this setting.
+# SOCIAL_AUTH_SAML_ENABLED_IDPS = {
+#     "testshib": {
+#         "entity_id": "https://idp.testshib.org/idp/shibboleth",
+#         "url": "https://idp.testshib.org/idp/profile/SAML2/Redirect/SSO",
+#         "x509cert": \
+#             """
+#             MIIDojCCAoqgAwIBAgIGAW0svbVqMA0GCSqGSIb3DQEBCwUAMIGRMQswCQYDVQQGEwJVUzETMBEG
+#             A1UECAwKQ2FsaWZvcm5pYTEWMBQGA1UEBwwNU2FuIEZyYW5jaXNjbzENMAsGA1UECgwET2t0YTEU
+#             MBIGA1UECwwLU1NPUHJvdmlkZXIxEjAQBgNVBAMMCXp1bGlwY2hhdDEcMBoGCSqGSIb3DQEJARYN
+#             aW5mb0Bva3RhLmNvbTAeFw0xOTA5MTMyGWRTffNaFw0yOTA5MTMyMjI4MTNaMIGRMQswCQYDVQQG
+#             EwJVUzETMBEGA1UECAwKQ2FsaWZvcm5pYTEWMBQGA1UEBwwNU2FuIEZyYW5jaXNjbzENMAsGA1UE
+#             CgwET2t0YTEUMBIGA1UECwwLU1NPUHJvdmlkZXIxEjAQBgNVBAMMCXp1bGlwY2hhdDEcMBoGCSqG
+#             SIb3DQEJARYNaW5mb0Bva3RhLmNvbTCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBAL3K
+#             E57ZDDVplrZO1RKpz9zFekhXZZFiPhW2TCTtoaI966sGaRCmV10cb1FCUxxI3ilcjY8G5irHYc5O
+#             D4S8+FeIaHb036VjtZZNCDkamE2zGZCix5wCpXhxhQrXkkPJbzO49hhDF3HHJJPwefGfYnPC8/Oj
+#             bZ0OUuR8KkNbgn2VnqwZtmb0EX5xrA+212UDyVQ7izVXOoBbvzeydLh8EWteEXjKBREKGBfCL9Kl
+#             x8JY7BlYrZx+13NeDQsL7bgTXMnTIp3MVP3xddRqsatwersRVGr9b/HzXxfwu/MU230swjsNlgLZ
+#             OXiYD43rNEkJRlFfMnlY8F3IoE1Mki2BJtMCAwEAATANBgkqhkiG9w0BAQsFAAOCAQEAI8Xk13NT
+#             rjo417U1+wZjUKqvB7iw4zGapMqpGRVavGw9hZmSCs8/AJAkFZQKMhAR9GGqf7JHHj/4fNEQ+XVh
+#             YF1jCUR/X2VwUiBseDHaUKj7EZiX9tIFEI/6LVfPRjKNy1RkEXHo7Lg4RnctclZ1KU7mIZkPSk1J
+#             fShKIUhtvNaCYJ4OVkN+giQQ6u9HwBqoBYikOBhvgXfIlBFD5H1n7JqxOjWZNO7Rhhx+TjD/0Dmd
+#             BE04J2bv5zCllBWSv2e1YFi+5SBTBq6FaIMz5c8T4WRFpRlnDEvvEREeAsAbaDiYvpJlO6hOzHNj
+#             KVmHpmQsDhsgXP02tDsfTARf3EXhbA==
+#             """.replace(' ', '').replace('\n', '').strip(),
+#         "attr_user_permanent_id": "email",
+#         "attr_first_name": "first_name",
+#         "attr_last_name": "last_name",
+#         "attr_username": "email",
+#         "attr_email": "email",
```

Comment:
If we wanted to sync additional attributes, can one just add them here, or does it only support that explicit list?

Reply:
Doesn't seem so, except fullname (which perhaps I should include here in the config). This is governed by ``social_details`` function in the pipeline, which gets the data sent by the IdP through ultimately this function:
```
    def get_user_details(self, attributes):
        """
        Given the SAML attributes extracted from the SSO response, get
        the user data like name.
        """
        return {
            'fullname': self.get_attr(attributes, 'attr_full_name',
                                      OID_COMMON_NAME),
            'first_name': self.get_attr(attributes, 'attr_first_name',
                                        OID_GIVEN_NAME),
            'last_name': self.get_attr(attributes, 'attr_last_name',
                                       OID_SURNAME),
            'username': self.get_attr(attributes, 'attr_username',
                                      OID_USERID),
            'email': self.get_attr(attributes, 'attr_email',
                                   OID_MAIL),
        }
```

which is defined in ``SAMLIdentityProvider`` in ``social_core.backends.saml``, so we don't have any nice means of controling it.

Reply:
Now that I think of it, I think:
```
#         "attr_user_permanent_id": "email",
#         "attr_fullname": "full_name",
#         "attr_username": "email",
#         "attr_email": "email",
```

is a more sensible default, since we use the 'fullname' value in the details dict in our part of the social pipeline, but not first name/last name.

Reply:
Although it seems that okta doesn't support sending the full name - only first and last name separately :thinking:

Reply:
Added a commit to handle that - if fullname isn't provided, ``social_associate_user_helper`` will try to construct it from first and last name attributes.